### PR TITLE
ci: change travis-ci to github-action

### LINKED ISSUE #9: Change travis-ci to github-action .

### PR DESCRIPTION
**Pull request by ffyuanda**
Fix: https://github.com/pycasbin/pymongo-adapter/issues/9

### CODE REVIEW

**Comment by ffyuanda** [comment]
a few points:

- a successful GitHub Actions run can be checked out at my [repo](https://github.com/ffyuanda/pymongo-adapter/actions/runs/941223520)
- the `deploy`section in the original `.travis.yml` file is replaced with the `Build package` and `Publish package` steps. The successful run of these two steps can be viewed [here](https://github.com/ffyuanda/pymongo-adapter/actions/runs/941148306), to test which **I have to comment out this** [line](https://github.com/ffyuanda/pymongo-adapter/blob/2457f53563570dc4f03f210a04dee47d941b0f73/.github/workflows/main.yml#L52). In the latest version, the line is not commented since we only want to publish the package when it is a new tagged release, namely realizing this [line](https://github.com/pycasbin/pymongo-adapter/blob/88b9b8a10c9401e2dded3995afb9c9d787564c59/.travis.yml#L23)

- I will recommend reset this [password](https://github.com/ffyuanda/pymongo-adapter/blob/d3cef6e02f483edd8488334f7871bc9f1c0dbed0/.github/workflows/main.yml#L56) part to the variable secret `${{ secrets.PYPI_API_TOKEN }}` for both safety and simplicity

> The secret used in `${{ secrets.PYPI_API_TOKEN }}` needs to be created on the settings page of your project on GitHub. See [Creating & using secrets](https://docs.github.com/en/actions/reference/encrypted-secrets).

**Comment by ffyuanda** [comment]
@hsluoyz @techoner @Zxilly plz review

**Comment by Zxilly** [comment]
Could you please use semantic release?

**Comment by ffyuanda** [comment]
@Zxilly plz check, I just used other repo's JSON file

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -88,11 +89,12 @@ jobs:
...
+      - name: Python Semantic Release
+        uses: relekang/python-semantic-release@master
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          pypi_token: ${{ secrets.PYPI_TOKEN }}
+#          PYPI_USERNAME: techlee
+#          PYPI_PASSWORD: pTL4bF82LZ+Usn7rhm/7tPDz4QiF+RWuVyu+7aNtgqy4JZ5nCloIxohSxODCr58vYabAMecxwCsvCVm7DSWeJiYzFiyWMiEIRxmuc6LHSNnG8pYIHuj9qBcOdyshQbN5ALpuvnpFX2sUQ20hy3ge6cdFT4xLZAm4nzmNnR6e20MNOT675agdNfgDsjnNEBV7lCGfvpYtAJgaB95f8P24Q/WX6JevcrhCLLoLOnZdjt8RxvhV95QF03E3dI9c7d1NwFEZ/IbK2WfXjZ8rY2vYutscSlvag2r+N/ZVlOwTQvwWeZHXkDu6bRsV2mKC1Sq2J6q6mwtFm81YLB5A0RgMIJu+4Ec36Xl1HxRCBgDxDh4WfZkx/yZJsAMqTKVnFsImDt76MD6sDyemG/cPhli3qkAYazgdL2rv9l8KQWVwIhUTN4zOigzDpoEGIvk3DjrAoCoyP/9U5oagBnHSdL/e5jHFoJQ9CS8RRUYvsez6QWE7V06qRLfTSpxmxzvFrA0RENw6RcpabVmbFac/HdMyv9aFL8Z/bWHf2vkyAYUDBDiyFKKYcGWxTsS3L+fTDvyycfy8NJ3jSgloJiXH2FFsEJ03ZxgCIONNMw8gZ58gglJC0jJHZbzjIAl7oNcm+6S0T56y3H6Hb2zQk0eRD6qiY6RQZowc+HH9Is0ejNhW9zc=
```
Comment (hsluoyz) [inline .github/workflows/main.yml]: We use GitHub secrets to store key, not plain text here.

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (ffyuanda) [reply .github/workflows/main.yml]: Sure, I just commented it out here cause I met some problems. Do we have pypi_token set up in the secrets?

Comment (ffyuanda) [reply .github/workflows/main.yml]: @techoner please? Do we have pypi_token implemented or only user and password?

Comment (hsluoyz) [reply .github/workflows/main.yml]: @ffyuanda use this:

image

Comment (ffyuanda) [reply .github/workflows/main.yml]: @hsluoyz it says 
image
so I don't think I can access it on my forked repo during a PR process, might need to merge this PR first and see if there are any issues...

**Comment by ffyuanda** [comment]
@Zxilly plz review

**Comment by Zxilly** [comment]
@ffyuanda semantic-release-pypi has some specific behaviors. plz refer to: https://github.com/pycasbin/flask-authz/pull/31, https://github.com/pycasbin/flask-authz/pull/30

**Comment by ffyuanda** [comment]
@Zxilly just removed the version in setup.py, this might resolve the no-release-on-pypi issue?

**Comment by Zxilly** [comment]
@ffyuanda also import `__version__` from setuptools, and add `setup.cfg`

File: .github/workflows/main.yml (hunk 1 above)
Comment (hsluoyz) [reply .github/workflows/main.yml]: @ffyuanda that's correct.

**Review by hsluoyz — APPROVED** [review]

**Comment by hsluoyz** [comment]
@ffyuanda squash into 1 commit

**Review by leeqvip — APPROVED** [review]

**Pull request merged**